Minor JavaScript technologies overview rewrite (#18586)
mdn/content: files/en-us/web/javascript/language_resources/index.md

---
{{JsSidebar}}

> **ECMAScript** is the scripting language that forms the basis of [JavaScript](/en-US/docs/Web/JavaScript). ECMAScript standardized by the [ECMA International](https://www.ecma-international.org/) standards organization in the **ECMA-262 and ECMA-402 specifications**.

Reviewer: ```suggestion
**ECMAScript** is the scripting language that forms the basis of [JavaScript](/en-US/docs/Web/JavaScript). ECMAScript is standardized by the [ECMA International](https://www.ecma-international.org/) standards organization in the **ECMA-262 and ECMA-402 specifications**.
```

> **ECMAScript** is the scripting language that forms the basis of [JavaScript](/en-US/docs/Web/JavaScript). ECMAScript standardized by the [ECMA International](https://www.ecma-international.org/) standards organization in the **ECMA-262 and ECMA-402 specifications**.

Reviewer 2: ```suggestion
**ECMAScript** is the scripting language that forms the basis of [JavaScript](/en-US/docs/Web/JavaScript). ECMAScript is standardized by the [ECMA International](https://www.ecma-international.org/) standards organization in the **ECMA-262 and ECMA-402 specifications**.
```

> The specifications are published on a yearly basis. The latest published version of ECMA-262 can be found in [PDF form](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-262.pdf) and [HTML form](https://262.ecma-international.org/13.0/). <https://tc39.es/ecma262/> and <https://tc39.es/ecma402/> host the current working drafts for ECMA-262 and ECMA-402.

Reviewer: ```suggestion
Snapshots of the specifications are published on a yearly basis. The current committee-approved snapshot of ECMA-262 can be found in [PDF form](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-262.pdf) and [HTML form](https://262.ecma-international.org/13.0/). ECMA-262 and ECMA-402 are continuously maintained and kept up to date by the specification editors; https://tc39.es/ecma262/ and https://tc39.es/ecma402/ host the latest, up-to-date versions.
```
I don’t feel strongly about the particular wording here — but I do feel that however it’s worded, it should be clear that https://tc39.es/ecma262/ and https://tc39.es/ecma402/ are in fact the latest versions of the spec, and that the ECMA-published versions are just snapshots (which are implicitly always out of date).

In other words, I think we want to try to make it clear to readers that they should be reading the https://tc39.es/ecma262/ and https://tc39.es/ecma402/ versions and really should not be reading the ECMA-published versions (because they otherwise might be missing a bug fix or typo fix, or some other refinement, etc., made to the spec/source after the current ECMA-approved version was published

Also, unless I’m missing something about the semantics of putting angle brackets around the tc39.es URLs, it doesn’t seem necessary…

Author: I agree—this seems better.

> Also, unless I’m missing something about the semantics of putting angle brackets around the tc39.es URLs, it doesn’t seem necessary…

Some Markdown renderers render it as plain text unless you explicitly mark it as a link. Seems to not be the case for us.

> The specifications are published on a yearly basis. The latest published version of ECMA-262 can be found in [PDF form](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-262.pdf) and [HTML form](https://262.ecma-international.org/13.0/). <https://tc39.es/ecma262/> and <https://tc39.es/ecma402/> host the current working drafts for ECMA-262 and ECMA-402.

Reviewer 2: Is there some kind of moving link we can use for this to stop it going out of date? Or some way we can grab BCD data for this?

Author: That's also my worry. The PDF form is auto-updating, but I don't know any for the HTML form. I personally just read https://tc39.es/ecma262/.

> Below are the historical editions:

Reviewer 3: This whole table below was strange and still is.

Either we list all editions in date order (or reverse), or we get rid of it.

I'd say, we get rid of it: it doesn't bring much (Web devs don't really care about the ES version, only if they can use it or not, and violate the _timeless_ principle: if we don't maintain it, it becomes naturally outdated (what is current will stop been current).

Author: Once we remove the table, this page will become too thin—shall we merge it with technologies overview?

Reviewer 3: Yes, I think so.
